app/models/atmosphere/appliance.rb

```diff
@@ -163,6 +163,12 @@ def assign_fund
         vmt.tenants.map(&:funds)
       end.flatten.uniq.compact
 
+      # Restrict to funds which belong to this appliance's owner
+      funds = funds & appliance_set.user.funds
```

Comment:
Maybe something like:

```ruby
funds.where(id: appliance_set.user.funds)
```

then most probably there will be 1 sql query instead of 2